### ISSUE TITLE
Energy condensers consume infinite emc from relay.

### ISSUE THREAD

**Issue by bigstiffy** [sinkillerj/ProjectE#1050]
On version 1.9.3h, if you connect multiple energy condensers to a MK3 anti-matter relay, the condensers appear to consume infinite emc when the condenser inventories are full.

I've left this running for a few hours, each condenser is set to produce low value items, with dumping stacks of diamonds, the help of a MK3 collector and the occasional boost booster of redmatter, emc has not started to accumulate in the relay.

**Comment by williewillus** [comment]
@MaPePeR your work? ;D

**Comment by MaPePeR** [comment]
Maybe ;) but it might also always been like that. The mk2 condensers did that with items, too, before I fixed that.

**Comment by MaPePeR** [comment]
I don't think they will consume infinite EMC... they will just fill their giant internal emc buffer that is nowhere shown in the GUI... (1.073.741.824 EMC buffer...)

**Comment by MaPePeR** [comment]
I noticed that the Condenser MK1 also has the faulty behavior we fixed in MK2: It always consumes items, even when it has a huge amount of emc stored and can not keep up with creating items... But when we make it stop consuming items it might fill up and will not have space to create new items...

Maybe change it to a "only consume items when we really need emc or there is no space left"?

**Comment by bigstiffy** [comment]
Yep, I was wrong. To satisfy the 3 billion+ emc buffer of the three condensers, it would take a little over 330 mk2 condensers fed into the MK3 relay. Had to set up a lot of pedestals with watch of flowing time to test that faster.
Will close since it's a non-issue then.

**Comment by MaPePeR** [comment]
I still think its kind of an issue. These should not store so much EMC when they don't need it.
@sinkillerj @williewillus @Lilylicious Thoughts?

---

My suggestion for the MK1 would be:

* If there is enough EMC stored to create an item
   * if there is space in the inventory: create the item
   * if there is no space in the inventory consume an item from the inventory (first/last/random or slot with lowest emc value?) and create item in that slot
* if there is not enough emc stored to create an item
   * Consume a single item from the inventory

And only accept up to the EMC value of the target item from adjacent emc providers.

This would kind of limit, but not totally prevent the MK1 from storing a lot of EMC.

-----

The MK2 would then work like this:

* Use all the stored emc to fill the output inventory
* When there is still space in the output inventory (so we used all our stored emc): 
  * Consume the content of the first non-empty input inventory slot   
     (Thats how it works at the moment, but I think we might also want to change that to 'consume up to 64 items' because: when we empty the first inventory slot its very likely that there will be only 1 new item in that first slot the next tick, but items might have stacked up in the remaining slots of the inventory - not sure how to handle items with different stacksizes, though.)

Because the MK2 uses all its stored EMC to create items we only stop accepting EMC from providers when the output inventory is full.

**Comment by williewillus** [comment]
It should be however it was in EE2/older versions of PE.
Because anything else would possibly break builds and we'd have rage

**Comment by MaPePeR** [comment]
Breaking existing builds is what happens when updating a mod. 
I would prefer to break/change the condensers  (once) instead of them being stupid forever. 

I  don't know how they worked in EE2. But the condensers being the biggest emc storage unit besides the transmutation table without them having any way to indicate such storage is bad. 

I don't even think they are supposed to store any emc.

**Comment by Lilylicious** [comment]
Regardless of if they should or should not store emc, they definitely should NOT store significant amounts of emc with no indication of it. I don't care if we limit the buffer, add a bar or even numbers to indicate the size of the buffer, three needs to be some way to tell what is happening.

If nothing else, we avoid more issues like this one.

**Comment by Heimdell** [comment]
How about to convert EMC representation to some BigInt?

**Comment by MaPePeR** [comment]
Integer size is not the problem here

**Comment by williewillus** [comment]
I stand by my position from months ago: just add a visual indicator to the gui.
